src/main/java/org/simlar/simlarserver/services/subscriberservice/SubscriberService.java

```diff
@@ -137,4 +141,12 @@ public List<SimlarId> filterSimlarIdsRegistered(final Collection<SimlarId> simla
                 .map(SimlarId::create)
                 .toList();
     }
+
+    public void deleteBySimlarId(final SimlarId simlarId) {
+        if (simlarId == null) {
+            throw new IllegalArgumentException("simlarId=" + null);
+        }
```

Comment:
> If a caller passes null in some parameter for which null values are prohibited, convention dictates that
> NullPointerException be thrown rather than IllegalArgumentException.
Effective Java, 3rd Edition, Item 72.

```suggestion
        Objects.requireNonNull(simlarId, "simlarId=" + null);
```